Source/CarthageKit/Project.swift

```diff
@@ -922,11 +913,9 @@ public final class Project { // swiftlint:disable:this type_body_length
 						continue
 					}
 
-					do {
-						try fileManager.createSymbolicLink(atPath: dependencyCheckoutURL.path, withDestinationPath: linkDestinationPath)
-					} catch let error as NSError {
-						return .failure(.writeFailed(dependencyCheckoutURL, error))
-					}
+					return Result(at: dependencyCheckoutURL, attempt: {
+						try fileManager.createSymbolicLink(atPath: $0.path, withDestinationPath: linkDestinationPath)
+					})
```

Comment:
⚠️ Accidentally introduced a functional change (moreover, a bug) by `return`ing the result’s value from 
 a for-loop — breaking the loop and returning from [the surrounding closure](https://github.com/jdhealy/Carthage/blob/ec4d59a44351ee56d545cb0446115a0c151dc3fa/Source/CarthageKit/Project.swift#L867-L922). Previously, [only the result’s error would break the for-loop](https://github.com/Carthage/Carthage/pull/2070/files#diff-3bd7f6198aa606c5ec6931e77cb3c42bL928).